crates/rrg/src/action/dump_process_memory.rs

```diff
@@ -340,6 +325,272 @@ mod linux {
     }
 }
 
+#[cfg(target_os = "windows")]
+mod windows {
+    use super::*;
+    use core::ffi::c_void;
+    use std::mem::MaybeUninit;
+    use windows_sys::Win32::Foundation::{CloseHandle, HANDLE};
+    use windows_sys::Win32::System::Diagnostics::Debug::ReadProcessMemory;
+    use windows_sys::Win32::System::Memory::{MEM_FREE, MEMORY_BASIC_INFORMATION, VirtualQueryEx};
+    use windows_sys::Win32::System::Threading::{OpenProcess, PROCESS_ALL_ACCESS};
+
+    struct ProcessHandle(HANDLE);
+
+    impl ProcessHandle {
+        fn open(pid: u32) -> std::io::Result<Self> {
+            // Open a handle to the process with all access rights
+            let handle: HANDLE = unsafe { OpenProcess(PROCESS_ALL_ACCESS, 0, pid) };
+            if handle.is_null() {
+                Err(std::io::Error::last_os_error())
+            } else {
+                Ok(Self(handle))
+            }
+        }
+    }
+
+    impl Drop for ProcessHandle {
+        fn drop(&mut self) {
+            unsafe { CloseHandle(self.0) };
+        }
+    }
+
+    pub struct MappedRegionIter {
+        process: ProcessHandle,
+        cur_addr: *mut c_void,
+    }
+
+    impl MappedRegionIter {
+        /// Creates a new [`MappedRegionIterator`] of the process with id `pid`.
+        pub fn from_pid(pid: u32) -> std::io::Result<Self> {
+            let process = ProcessHandle::open(pid)?;
+            Ok(Self {
+                process,
+                cur_addr: std::ptr::null_mut(),
+            })
+        }
+    }
+
+    fn parse_permissions(mem_info: &MEMORY_BASIC_INFORMATION) -> Permissions {
+        use windows_sys::Win32::System::Memory::*;
+
+        // Info obtained from https://learn.microsoft.com/en-us/windows/win32/memory/memory-protection-constants
+        // and https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-memory_basic_information
+
+        const EXECUTE_FLAGS: u32 =
+            PAGE_EXECUTE | PAGE_EXECUTE_READ | PAGE_EXECUTE_READWRITE | PAGE_EXECUTE_WRITECOPY;
+        const WRITE_FLAGS: u32 =
+            PAGE_EXECUTE_READWRITE | PAGE_EXECUTE_WRITECOPY | PAGE_READWRITE | PAGE_WRITECOPY;
+
+        // Use the protection information of the actual committed memory if available,
+        // otherwise the flags with which the memory mapping was created.
+        let flags = if mem_info.State == MEM_COMMIT {
+            mem_info.Protect
+        } else {
+            mem_info.AllocationProtect
+        };
+
+        let execute = flags & EXECUTE_FLAGS != 0;
+        let write = flags & WRITE_FLAGS != 0;
+        // The only flag which disables write access is PAGE_NOACCESS
+        let read = flags & PAGE_NOACCESS == 0;
+        let private = mem_info.Type == MEM_PRIVATE;
+        Permissions {
+            read,
+            write,
+            execute,
+            private,
+            shared: false,
+        }
```

Comment:
See, this is exactly what I was ranting about the other day regarding `Foo { bar }` syntax! :D Why write it like this instead of just:

```rust
Permissions {
    read: flags & PAGE_NOACCESS == 0,
    write: flags & WRITE_FLAGS != 0,
    execute: flags & EXECUTE_FLAGS != 0,
    private: mem_info.Type == MEM_PRIVATE,
    shared: false,
}
```

The syntax conditions you towards using `let`, the urge is strong, I know.